core/trino-main/src/main/java/io/trino/sql/planner/QueryPlanner.java

```diff
@@ -701,6 +721,45 @@ public PlanNode plan(Update node)
         return createMergePipeline(table, relationPlan, projectNode, rowIdSymbol, mergeRowSymbol);
     }
 
+    private static Expression rewriteToAssignment(Map<String, Expression> columnNameToAssignment, Expression sourceExpression)
+    {
+        return ExpressionTreeRewriter.rewriteWith(
+                new ExpressionRewriter<Void>()
+                {
+                    @Override
+                    public Expression rewriteIdentifier(Identifier node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
+                    {
+                        if (columnNameToAssignment.containsKey(node.getValue().toLowerCase(ENGLISH))) {
+                            return columnNameToAssignment.get(node.getValue().toLowerCase(ENGLISH));
+                        }
```

Comment:
The planner shouldn't be doing any identifier-based resolution. Resolving and matching identifiers involves understanding delimited/non-delimited identifier matching rules and is the job of the analyzer. The analyzer should then record whichever annotations or supporting structures necessary for the planner to do its job without looking at identifier semantics.

Also, we're in the process of decoupling the AST from the IR, so it will no longer be possible to rewrite expressions in this manner as the AST domain and IR domain will be separate hierarchies.

Reply:
Moved to `StatementAnalyzer` and changed to rewrite using `Identifier` instead of String column name. Please take another look.